extended/src/main/java/io/kubernetes/client/extended/workqueue/ratelimiter/ItemExponentialFailureRateLimiter.java

```diff
@@ -0,0 +1,51 @@
+package io.kubernetes.client.extended.workqueue.ratelimiter;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * ItemExponentialFailureRateLimiter does a simple baseDelay*10<sup>num-failures</sup> limit dealing
+ * with max failures and expiration are up to the caller
+ */
+public class ItemExponentialFailureRateLimiter<T> implements RateLimiter<T> {
+
+  private Duration baseDelay;
+  private Duration maxDelay;
+
+  private Map<T, Integer> failures;
+
+  public ItemExponentialFailureRateLimiter(Duration baseDelay, Duration maxDelay) {
+    this.baseDelay = baseDelay;
+    this.maxDelay = maxDelay;
+
+    failures = new HashMap<>();
+  }
+
+  @Override
+  public synchronized Duration when(T item) {
```

Comment:
```suggestion
  @Override
  public synchronized Duration when(T item) {
    int exp = failures.getOrDefault(item, 0);
    failures.put(item, exp + 1);
    long d = maxDelay.toMillis() >> exp;
    return d > baseDelay.toMillis() ? baseDelay.multipliedBy(1 << exp) : maxDelay;
  }
```

how do you think about this approach?

Reply:
IMHO, Java has many built-in thread-safe data structure, try to avoid using synchronized method here, `java.util.concurrent.ConcurrentHashMap` should be a good fit for this case, and is much more high efficient.

Reply:
I think `ConcurrentHashMap` is not thread-safe when we are doing multiple operations...

Reply:
You can use a combination of `ConcurrentHashMap` and `AtomicInteger` here.

Reply:
OK, replaced with AtomicLongMap now.